canonical-webteam/global-nav
# Optimise and encapsulate global-nav scss

## Done

- Reduced the amount of Vanilla dependencies to reduce the built file size (this includes removing the grid in favour of flex containers)
- Added the option to pass a config object to the `createNav()` function, so you can specify the max-width of the rows
- Renamed the global nav class names to be more descriptive
- Updated the README
- Fixed sass and JS linting errors

## QA

- Pull code and run `yarn lint` and check there are no errors
- Run `yarn watch`, wait for the file to be built and open `index.html`
- Check that the global nav functions pretty much exactly the same as the one on mongoose.ubuntu.com (I've had to make some minor changes to reduce the Vanilla dependencies)
- Remember to check in different browsers and viewpoint sizes
- Open `index.html` and pass an object with maxWidth property to the `createNav()` function:
```js
canonicalGlobalNav.createNav({
  maxWidth: '50rem'
});
```
- Check that the max-width of the global nav changes accordingly
- For extra points you can try including `dist/index.js` in different projects and running `canonicalGlobalNav.createNav()` to see how it looks

## Issues

Fixes ubuntudesign/web-squad#748

## Screenshots

Testing on the new mir-server.io homepage:

![0 0 0 0_8028_](https://user-images.githubusercontent.com/25733845/43842606-a41f3eec-9b1d-11e8-818f-44cb30c1a46d.png)
